### PR TITLE
New Adapter: Zentotem

### PR DESCRIPTION
**Pull request by zentotem**
Docs: https://github.com/prebid/prebid.github.io/pull/5553

### CODE REVIEW

**Comment by zentotem** [comment]
Hello! Apologies for making some changes during the review process, but I had to change “getMediaTypeForBid”

File: adapters/zentotem/zentotem.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	if request == nil {
+		return nil, []error{&errortypes.BadInput{Message: "empty request"}}
+	}
+	if len(request.Imp) == 0 {
+		return nil, []error{&errortypes.BadInput{Message: "empty request.Imp"}}
+	}
```
Comment (gargcreation1992) [inline adapters/zentotem/zentotem.go:32]: No need to perform these checks. These are covered by the core predid server code.

File: adapters/zentotem/zentotem.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+			Uri:    a.endpoint,
+			Body:   requestJSON,
+			ImpIDs: []string{imp.ID},
+		}
+		requests = append(requests, requestData)
+	}
+	return requests, errors
+}
+
+func getMediaTypeForBid(imp openrtb2.Imp) (openrtb_ext.BidType, error) {
```
Comment (gargcreation1992) [inline adapters/zentotem/zentotem.go:58]: Requesting you to please re-write this function to use MType field.

File: adapters/zentotem/zentotem.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+		err := &errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info.", responseData.StatusCode),
+		}
+		return nil, []error{err}
+	}
+
+	impMap := map[string]openrtb2.Imp{}
+	for _, imp := range request.Imp {
+		impMap[imp.ID] = imp
+	}
```
Comment (gargcreation1992) [inline adapters/zentotem/zentotem.go:98]: You should not be creating the impMap if you use Mtype field set in the bid response.

File: static/bidder-info/zentotem.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+endpoint: "https://rtb.zentotem.net/bid?sspuid=cqlnvfk00bhs0b6rci6g"
+maintainer:
+  email: support@zentotem.net
```
Comment (gargcreation1992) [inline static/bidder-info/zentotem.yaml:3]: sent email for verification.

File: adapters/zentotem/zentotem.go (hunk 1 above)
Comment (zentotem) [reply adapters/zentotem/zentotem.go]: ### fixed

File: adapters/zentotem/zentotem.go (hunk 2 above)
Comment (zentotem) [reply adapters/zentotem/zentotem.go]: fixed

File: adapters/zentotem/zentotem.go (hunk 3 above)
Comment (zentotem) [reply adapters/zentotem/zentotem.go]: fixed

File: static/bidder-info/zentotem.yaml (hunk 4 above)
Comment (zentotem) [reply static/bidder-info/zentotem.yaml:3]: we sent a response letter for verification

**Comment by zentotem** [comment]
@gargcreation1992
Hello! I fixed all the comments you left above

**Review by gargcreation1992 — DISMISSED** [review]

**Comment by bsardo** [comment]
Hi @zentotem, we recently released PBS 3.0, more specifically v3.1.0, which updates Prebid Server package import references throughout the project from `v2` to `v3`.
For example:
```
import (
    "github.com/prebid/prebid-server/v3/adapters"
)
```
As a result, please merge with master (no rebase) and then ensure all Prebid Server package import references in the files you’ve changed are `v3` such that the test suite passes so we can resume reviewing. Thanks!

**Pull request closed without merging**